pkg/util/constraint_cache.go

```diff
@@ -0,0 +1,24 @@
+package util
+
+import (
+	"sync"
+)
+
+type ConstraintsCache struct {
+	CacheMux sync.RWMutex
```

Comment:
nit: should be a hidden variable

Reply:
do you mean anonymous? updated

Reply:
Sorry, "private" would have been better than "hidden". It should be a private variable so that outside packages can't muck with the internals, bypassing the lock.

Reply:
oh i see. mutex should be private now, but cache is used by `constrainttemplate` package

Reply:
It looks like mutex is just anonymous (not private), unless I missed a change?

No external package should use the `cache` member directly b/c they can then bypass the mutex.

Reply:
as discussed, cache should be private now